modules/account-lib/src/coin/stacks/keyPair.ts

```diff
@@ -0,0 +1,95 @@
+import { DefaultKeys, isPrivateKey, isPublicKey, isSeed, KeyPairOptions } from '../baseCoin/iface';
+import { Secp256k1ExtendedKeyPair } from '../baseCoin/secp256k1ExtendedKeyPair';
+import { isValidPrivateKey, isValidPublicKey } from './utils';
+
+const DEFAULT_SEED_SIZE_BYTES = 64;
+
+export class KeyPair extends Secp256k1ExtendedKeyPair {
+  /**
+   * Public constructor. By default, creates a key pair with a random master seed.
+   *
+   * @param { KeyPairOptions } source Either a master seed, a private key, or a public key
+   */
+  constructor(source?: KeyPairOptions) {
+    super(source);
+    if (!source) {
+      const seed = randomBytes(DEFAULT_SEED_SIZE_BYTES);
+      this.hdNode = HDNode.fromSeedBuffer(seed);
+    } else if (isSeed(source)) {
+      this.hdNode = HDNode.fromSeedBuffer(source.seed);
+    } else if (isPrivateKey(source)) {
+      this.recordKeysFromPrivateKey(source.prv);
+    } else if (isPublicKey(source)) {
+      this.recordKeysFromPublicKey(source.pub);
+    } else {
+      throw new Error('Invalid key pair options');
+    }
+
+    if (this.hdNode) {
+      this.keyPair = this.hdNode.keyPair;
+    }
+  }
+
+  /**
+   * Build an ECPair from a private key.
+   *
+   * The private key is either 32 or 33 bytes long (64 or 66 characters hex).
+   * If it is 32 bytes long, set the keypair's "compressed" field to false.
+   * A 33 byte key has 0x01 as the last byte.
+   *
+   * @param {string} prv A raw private key
+   */
+  recordKeysFromPrivateKey(prv: string): void {
+    if (!isValidPrivateKey(prv)) {
+      throw new Error('Unsupported private key');
+    }
+
+    const compressed = prv.length === 66;
+    this.keyPair = ECPair.fromPrivateKeyBuffer(Buffer.from(prv.slice(0, 64), 'hex'));
+    this.keyPair.compressed = compressed;
+  }
+
+  /**
+   * Build an ECPair from a public key.
+   *
+   * The public key is either 33 or 65 bytes long (66 or 130 characters in hex),
+   * depending on the compression.
+   *
+   * If it is 32 bytes long, set the keypair's "compressed" field to false.
+   * A 33 byte key has 0x01 as the last byte.
+   *
+   * @param {string} pub A raw private key
+   */
+  recordKeysFromPublicKey(pub: string): void {
+    if (!isValidPublicKey(pub)) {
+      throw new Error('Unsupported public key');
+    }
+
+    this.keyPair = ECPair.fromPublicKeyBuffer(Buffer.from(pub, 'hex'));
+  }
+
+  /**
+   * Stacks default keys format is raw private and uncompressed public key
+   *
+   * @param {boolean} compressed - Compress public key (defaults to false)
+   * @returns {DefaultKeys} The keys in the protocol default key format
+   */
+  getKeys(compressed = false): DefaultKeys {
+    return {
+      pub: this.keyPair.Q.getEncoded(compressed).toString('hex'),
+      prv: this.keyPair.d ? this.keyPair.d.toBuffer(32).toString('hex') : undefined,
+    };
+  }
+
+  /**
+   * Get a public address of an uncompressed public key.
+   *
+   * @returns {string} The public address
+   */
+  getAddress(): string {
```

Comment:
merits in allowing `compressed` to be an optional variable here?

Reply:
`getAddress` in the parent class takes an optional `AddressFormat` -- a string that's either `hex` or `base58`; adding support for a boolean for `compressed` breaks that signature.

Perhaps this function should be something like

```
import { getAddressFromPublicKey, TransactionVersion } from '@stacks/transactions';

  /**
   * Get a public address of an uncompressed public key.
   *
   * @param {boolean} compressed - Compress public key (defaults to false)
   * @param {TransactionVersion} network - select Mainnet or Testnet for the address
   * @returns {string} The public address
   */
  getAddress(compressed = false, network = TransactionVersion.Mainnet): string {
    return getAddressFromPublicKey(this.getKeys(compressed).pub, network);
  }
```

This lets us generate addresses for both the Stacks mainnet and testnet.  The address encodes the network [internally](https://github.com/stacksgov/sips/blob/main/sips/sip-005/sip-005-blocks-and-transactions.md#accounts) and in string format using the [account generation rules](https://github.com/stacksgov/sips/blob/main/sips/sip-005/sip-005-blocks-and-transactions.md#transaction-authorization): `SP...` for single-signature transactions on the mainnet and `ST...` on the testnet.

Reply:
Yes please, let's go with this approach

Reply:
I could not figure out how to reuse the `getAddress` type signature in the base class, so I created a separate function for this.